store/chat.go

```diff
@@ -0,0 +1,60 @@
+package store
```

Comment:
This package shouldn't be called directly from services, you need to use `interface`, because:
* at any time we can start using another database, and this should not force us to change types or code in the service code.
* when you write unit tests for services, they should never make real requests to DB, only to Mock 

store/store.go - describe interface, input/output struct
store/badger/badger.go - this is where this file should be, services don't need to know nothing about package `store/badger`

Reply:
Do I understand correctly that `store/badger/chat.go` should implement the interface from `store/store.go`?

Reply:
Correct

Reply:
I think it's not a good practice when lower-level packages refer to higher-level ones.

Reply:
You're wrong. With interfaces when may be several implementation this is widespread practice. It doesn't matter what is referring to. The main thing is `badger` is **implementation** of the **store**.
You can keep `chat.go` in `/store` folder this is also OK, but it is better to separate _interfaces_ from _drivers_. because if over time you need to add drivers to other databases, you will have to pilled up all in one package `/store`